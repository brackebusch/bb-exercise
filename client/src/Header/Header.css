.Header {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 29px;
}

.Date-Title {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Title {
  font-weight: 400;
  font-size: 32px;
  line-height: 32px;
  text-size-adjust:100%;
  text-rendering:optimizeLegibility;
  margin-left: 32px;
}

.Date {
  background-color: #f3f3f3;
  height: 70px;
  width: 70px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-radius: 3px;
} 

.Month {
  text-transform: uppercase;
  font-weight: 700;
  letter-spacing: 1.5px;
  font-size: 14px;
  text-align: center;
}

.Day {
  font-size: 28px;
  letter-spacing: 2px;
  text-align: center;
}

img {
  height: 32px;
  width: 225px;
  align-self: flex-start;
  margin: 15px 0px;
}