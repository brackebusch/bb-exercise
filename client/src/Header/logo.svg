<!-- <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3"> -->
<svg width="196" height="29" class="dib v-mid" viewBox="0 0 196 29" xmlns="http://www.w3.org/2000/svg">
    <g fill="#00A1DF">
        <path d='M 5.136 1.403 l 0.384 0.047 l 0.096 0.234 c 0.864 0.047 1.776 -0.187 1.776 -0.187 c 0.096 -0.094 0.336 -0.374 0.096 -0.515 c -0.96 0.187 -2.352 -0.14 -2.16 -0.047 c -0.192 -0.046 -0.432 -0.187 -0.672 -0.42 c -0.432 -0.702 -0.672 -0.234 -0.672 0 c -0.048 0.841 0.192 0.795 1.152 0.888 Z M 11.76 13.144 a 6.556 6.556 0 0 0 -0.192 -1.03 c -0.48 -1.543 -1.056 -3.18 -2.64 -3.882 c -0.672 -0.234 -1.92 -0.187 -2.112 -1.73 V 1.87 l -2.016 0.42 v 0.282 l -0.048 0.888 v 0.936 c 0.048 2.385 0.192 3.882 -2.496 4.864 c -1.632 0.608 -1.728 1.31 -1.92 3.134 c 0 0.14 -0.048 0.28 -0.048 0.374 c 0 -0.046 -0.528 5.146 1.248 13.378 c 0 0 0.288 1.356 0.288 1.543 c 0 0.094 0.144 0.234 0.384 0.421 c 3.024 0.702 3.696 0.187 5.28 0.421 c 1.632 0.187 3.072 -0.795 3.072 -0.842 c 0.144 -0.046 0.672 -1.122 0.768 -2.432 c 0.72 -3.602 0.576 -9.355 0.432 -12.114 Z'/>
    </g>
    <text x="21" y="18" fill="#555" style="font-size: 14px;font-weight: 600;letter-spacing: 1.6px;" textLength="175">BLUE BOTTLE COFFEE</text>
</svg>