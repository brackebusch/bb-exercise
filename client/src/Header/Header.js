import React from 'react';
import logo from './logo.svg';
import moment from 'moment'; 
import './Header.css';

const Header = ({changeModalState}) => {
  return(
    <div>
      <img src={logo} className='App-logo' alt='logo' />
      <div className='Header'>
        <div className='Date-Title'>
          <div className='Date'>
            <span className='Month'>{moment().format('MMM')}</span>
            <span className='Day'>{moment().format('DD')}</span>
          </div>
          <span className='Title'>Perfectly Ground Work Orders</span>
        </div>
        <button onClick={changeModalState}>CREATE ORDER</button>
      </div>
    </div>
  );
}

export default Header