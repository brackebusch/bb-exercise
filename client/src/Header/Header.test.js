import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import moment from 'moment'

  const changeModalState = jest.fn()

it('renders without crashing', () => {
  const wrapper = shallow(
    <Header changeModalState={changeModalState} />
  );

  expect(wrapper).toMatchSnapshot()
});

it("contains today's date", () => {
  const wrapper = mount(
      <Header changeModalState={changeModalState} />
    );

  const text = wrapper.find('div.Date').text();

  expect(text).toContain(moment().format('MMMDD'));
});

it("calls changeModalState on button click", () => {
  const wrapper = mount(
    <Header changeModalState={changeModalState} />
  );

  wrapper
    .find("button")
    .simulate("click");

  expect(changeModalState).toBeCalled();
});