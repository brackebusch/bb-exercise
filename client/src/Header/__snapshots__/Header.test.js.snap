// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    changeModalState={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />,
        <div
          className="Header"
        >
          <div
            className="Date-Title"
          >
            <div
              className="Date"
            >
              <span
                className="Month"
              >
                Oct
              </span>
              <span
                className="Day"
              >
                10
              </span>
            </div>
            <span
              className="Title"
            >
              Perfectly Ground Work Orders
            </span>
          </div>
          <button
            onClick={[MockFunction]}
          >
            CREATE ORDER
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "App-logo",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Date-Title"
            >
              <div
                className="Date"
              >
                <span
                  className="Month"
                >
                  Oct
                </span>
                <span
                  className="Day"
                >
                  10
                </span>
              </div>
              <span
                className="Title"
              >
                Perfectly Ground Work Orders
              </span>
            </div>,
            <button
              onClick={[MockFunction]}
            >
              CREATE ORDER
            </button>,
          ],
          "className": "Header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Date"
                >
                  <span
                    className="Month"
                  >
                    Oct
                  </span>
                  <span
                    className="Day"
                  >
                    10
                  </span>
                </div>,
                <span
                  className="Title"
                >
                  Perfectly Ground Work Orders
                </span>,
              ],
              "className": "Date-Title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="Month"
                    >
                      Oct
                    </span>,
                    <span
                      className="Day"
                    >
                      10
                    </span>,
                  ],
                  "className": "Date",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Oct",
                      "className": "Month",
                    },
                    "ref": null,
                    "rendered": "Oct",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "10",
                      "className": "Day",
                    },
                    "ref": null,
                    "rendered": "10",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Perfectly Ground Work Orders",
                  "className": "Title",
                },
                "ref": null,
                "rendered": "Perfectly Ground Work Orders",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CREATE ORDER",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "CREATE ORDER",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />,
          <div
            className="Header"
          >
            <div
              className="Date-Title"
            >
              <div
                className="Date"
              >
                <span
                  className="Month"
                >
                  Oct
                </span>
                <span
                  className="Day"
                >
                  10
                </span>
              </div>
              <span
                className="Title"
              >
                Perfectly Ground Work Orders
              </span>
            </div>
            <button
              onClick={[MockFunction]}
            >
              CREATE ORDER
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "App-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Date-Title"
              >
                <div
                  className="Date"
                >
                  <span
                    className="Month"
                  >
                    Oct
                  </span>
                  <span
                    className="Day"
                  >
                    10
                  </span>
                </div>
                <span
                  className="Title"
                >
                  Perfectly Ground Work Orders
                </span>
              </div>,
              <button
                onClick={[MockFunction]}
              >
                CREATE ORDER
              </button>,
            ],
            "className": "Header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="Date"
                  >
                    <span
                      className="Month"
                    >
                      Oct
                    </span>
                    <span
                      className="Day"
                    >
                      10
                    </span>
                  </div>,
                  <span
                    className="Title"
                  >
                    Perfectly Ground Work Orders
                  </span>,
                ],
                "className": "Date-Title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="Month"
                      >
                        Oct
                      </span>,
                      <span
                        className="Day"
                      >
                        10
                      </span>,
                    ],
                    "className": "Date",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Oct",
                        "className": "Month",
                      },
                      "ref": null,
                      "rendered": "Oct",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "10",
                        "className": "Day",
                      },
                      "ref": null,
                      "rendered": "10",
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Perfectly Ground Work Orders",
                    "className": "Title",
                  },
                  "ref": null,
                  "rendered": "Perfectly Ground Work Orders",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CREATE ORDER",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "CREATE ORDER",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
